### ISSUE TITLE
'vpx/vpx_encoder.h' file not found

### ISSUE THREAD

**Issue by kaida1234** [drawpile/Drawpile#828]
Hi,I followed the Instructions(https://github.com/drawpile/Drawpile/wiki/Building-from-sources) to compile，but it didn't work.After cmake it seems normal,but after make it doesn't work.
![image](https://user-images.githubusercontent.com/44886124/68749090-9d95d900-0638-11ea-8d5a-2147dcf1deee.png)
![image](https://user-images.githubusercontent.com/44886124/68749158-b9997a80-0638-11ea-8d6a-455b067edd6b.png)
![image](https://user-images.githubusercontent.com/44886124/68749181-c322e280-0638-11ea-8dae-d5ebb8574db4.png)
Is it where I configured Incorrectly？Help!!!
Thanks a lot.

**Comment by kaida1234** [comment]
the Instruction I followed is from https://github.com/drawpile/Drawpile/wiki/Building-from-sources

**Comment by callaa** [comment]
Looks like I haven't documented how to build the libvpx package. How did you install it? My first guess is that the installation is incomplete.

**Comment by kaida1234** [comment]
> Looks like I haven't documented how to build the libvpx package. How did you install it? My first guess is that the installation is incomplete.

Thank you for your answer.

I followed the instructions(https://github.com/drawpile/Drawpile/wiki/Building-from-sources) to compile.
Firstly, I install cmake-3.15.5-Darwin-x86_64.dmg, qt-opensource-mac-x64-5.10.0.dmg,Visual Studio (community edition),Xcode and Xcode command line tools.
Secondly, I run the script pkg/mac/install-deps.sh.
Lastly, cmake .. -DCMAKE_PREFIX_PATH=/Users/admin/Qt5.10.0/5.10.0/clang_64 and make.
I didn't install libvpx package, because after the second step,  I saw pkg/mac/deps/libvpx-1.8.0/vpx/vpx_encoder.h. Even so, it didn't work after make. I have no idea what is wrong with it.

**Comment by callaa** [comment]
If you ran the install-deps script, then the libvpx package should have been installed properly. 

Visual studio is only needed for windows development and in any case the mac version can't be used for C++ applications. (Huh, I didn't know they even had a mac version at all.)

Is there a specific reason you're using Qt 5.10 and not the latest LTS release (5.12.6)? Unless you need that version (e.g. it's the last version to support your macOS version,) I recommend using the newer one. (Also, did you use the installer utility or a package that install a specific version? I'm not sure it makes any difference, but the installer is very convenient.)

You can check if the libvpx include files are where they should be by looking at `/Users/admin/Qt5.10.0/5.10.0/clang_64/include`. There should be a `vpx` subdirectory there that contains vpx_encoder.h.
You can also try deleting the libvpx directory from the `deps` directory, then rerunning the install script.

**Comment by kaida1234** [comment]
> If you ran the install-deps script, then the libvpx package should have been installed properly.
> 
> Visual studio is only needed for windows development and in any case the mac version can't be used for C++ applications. (Huh, I didn't know they even had a mac version at all.)
> 
> Is there a specific reason you're using Qt 5.10 and not the latest LTS release (5.12.6)? Unless you need that version (e.g. it's the last version to support your macOS version,) I recommend using the newer one. (Also, did you use the installer utility or a package that install a specific version? I'm not sure it makes any difference, but the installer is very convenient.)
> 
> You can check if the libvpx include files are where they should be by looking at `/Users/admin/Qt5.10.0/5.10.0/clang_64/include`. There should be a `vpx` subdirectory there that contains vpx_encoder.h.
> You can also try deleting the libvpx directory from the `deps` directory, then rerunning the install script.

Thank you for your respond. It is too busy these days.

I tried to compile it all over again on another Mac OS computer last Friday, and it works!!!

Firstly, I install cmake-3.15.5-Darwin-x86_64.dmg, Xcode and Xcode command line tools.
Secondly, I brew install qt instead of installing qt-opensource-mac-x64-5.10.0.dmg(After I installed qt-opensource-mac-x64-5.10.0.dmg and ran the script pkg/mac/install-deps.sh, I wrote a HelloWorld program on /Users/admin/Qt5.10.0/Qt\ Creator.app, it didn't work. However, it worked before that. For this reason, I decide brew install qt instead of installing qt-opensource-mac-x64-5.10.0.dmg)
Thirdly, I run the script pkg/mac/install-deps.sh.
Lastly, cmake .. -DCMAKE_PREFIX_PATH=/usr/local/Cellar/qt/5.13.0 and make.
And...It works!!!
Maybe there's something wrong with the development environment of my first computer.

![image](https://user-images.githubusercontent.com/44886124/69222598-c24bfc80-0bb4-11ea-8cb3-3278be60120c.png)

And now, I want to run Drawpile on Xcode, so:
Firstly, sudo xcode-select --switch /Applications/Xcode.app/ .
Secondly, cmake .. -DCMAKE_PREFIX_PATH=/usr/local/Cellar/qt/5.13.0 -G "Xcode".
After that, I get a file /Users/ht/drawpile/Drawpile/build/drawpile.xcodeproj .
Thirdly, I open /Users/ht/drawpile/Drawpile/build/drawpile.xcodeproj with Xcode, and then I press the button with the triangle icon in the upper left corner to build and run Drawpile.
But...It doesn't work as well as I thought.

![image](https://user-images.githubusercontent.com/44886124/69227014-1d352200-0bbc-11ea-9e44-5062b14ae5ec.png)

![image](https://user-images.githubusercontent.com/44886124/69227685-38ecf800-0bbd-11ea-8060-13ba0e17d918.png)

![image](https://user-images.githubusercontent.com/44886124/69228186-160f1380-0bbe-11ea-900b-953d321962e0.png)

In fact, there is no such file or directory. I have no idea what is wrong with it once again...

I just want to run and set breakpoints to debug  the code related to the drawing function.

By the way, Is there any logs in the code of Drawpile? As far as I know, there's no log.

Thanks a lot in advance!!

**Comment by callaa** [comment]
Hmm, I haven't used Xcode very much at all, so I'm not sure what could be wrong. If you can get Qt Creator working, it has a debugger too.

**Comment by kaida1234** [comment]
> Hmm, I haven't used Xcode very much at all, so I'm not sure what could be wrong. If you can get Qt Creator working, it has a debugger too.

Oh, thank you. 

After I opened the root CMakeLists.txt file with Qt Creator and press Build, I got Drawpile working and found logs in /Users/ht/Library/Application Support/drawpile/drawpile/logs/. And I am reading the codes in Qt Creator now. Thank you.

![image](https://user-images.githubusercontent.com/44886124/69477109-2bce4400-0e1d-11ea-89e8-77f88f639831.png)

But, when I press to start debugging, It show me "Unable to create a debugging engine of the type "No engine". Maybe it's because I brew install qt and downloaded qt-creator-opensource-mac-x86_64-4.4.1.dmg, but not downloaded and installed qt-opensource-mac-x64-5.13.2.dmg ?? :(

![image](https://user-images.githubusercontent.com/44886124/69477131-6932d180-0e1d-11ea-97d9-559961b264fc.png)

![image](https://user-images.githubusercontent.com/44886124/69477152-9bdcca00-0e1d-11ea-8a62-32324b274d0e.png)
